### ISSUE TITLE
Getting Uncaught TypeError

### ISSUE THREAD

**Issue by Thirunav** [craftpip/jquery-confirm#276]
When I was use this plugin getting an error 
"Uncaught TypeError: a(...).on is not a function
    at jquery-confirm.min.js:10
    at jquery-confirm.min.js:10"

My code is :

<link type="text/css" rel="stylesheet" href="/css/main.css" />
<link type="text/css" rel="stylesheet" href="/css/jquery.tabs.css" />
<link type="text/css" rel="stylesheet" href='/css/autocomplete.css' />
<link type="text/css" rel="stylesheet" href='/css/jquery.ui.theme.css' />
<link type="text/css" rel="stylesheet"
	href='/css/jquery.ui.autocomplete.css' />
<link type="text/css" rel="stylesheet" href='/css/jquery.ui.core.css' />
<link type="text/css" rel="stylesheet" href='/css/ui.jqgrid.css' />
<link rel="stylesheet" href="/css/facebox.css" type="text/css" />
<link rel="shortcut icon" href="/images/favicon.ico" />

<script src='/scripts/jquery-1.4.2.min.js'></script>
<script src='/scripts/jquery.bgiframe.min.js'></script>
<script src='/scripts/jquery.livequery.js'></script>
<script src='/scripts/jquery-ui-1.8.custom.js'></script>
<script src='/scripts/grid.locale-en.js'></script>
<script src='/scripts/jquery.jqGrid.min.js'></script>
<script src='/scripts/advancedsearch.js'></script>
<script src="/scripts/facebox.js" type="text/javascript"></script>
<script src="/scripts/blockKeyPress.js" type="text/javascript"></script>

<link type="text/css" rel="stylesheet" href='/css/jquery.ui.dialog.css' />
<script src='/scripts/jquery-ui-1.8.custom.js'></script>

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/jquery-confirm/3.2.0/jquery-confirm.min.css">
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-confirm/3.2.0/jquery-confirm.min.js"></script>

<script type='text/javascript'>
function open(){
	$.alert({
	    title: 'Alert!',
	    content: 'Simple alert!',
	});
}

$(document).ready(function(){
		var isDefaultLanding = "<%=request.getSession().getAttribute("isDefaultLanding")%>";
		if (isDefaultLanding == 'Y') {
			
	    } else {
	    	return false;
	    }
		//CustomerService.fetchNewsAndAnnouncements(callMetaData);
		<%request.getSession().removeAttribute("isDefaultLanding");%>
	});

</script>

**Comment by craftpip** [comment]
please use the non minified version of the file, 
i.e. https://cdnjs.cloudflare.com/ajax/libs/jquery-confirm/3.2.0/jquery-confirm.js
and let me know the line number of the error.

**Comment by Thirunav** [comment]
I used the minified version, but still the same problem is not resolved

In the below line is throwing the error Uncaught TypeError $.alert is not a function

$.alert({ title: 'Alert!', content: 'Simple alert!', });

**Comment by craftpip** [comment]
I will require the line number from jquery-confirm.js to determine the problem,

**Comment by craftpip** [comment]
"Uncaught TypeError: a(...).on is not a function
at jquery-confirm.min.js:10
at jquery-confirm.min.js:10"

Here the line number is 10, that is hard to debug as its a minified version.

**Comment by emersonajr** [comment]
Thirunav, there is a lost comma after Simple Alert... Try deleting it. It will be like:
$.alert({ title: 'Alert!', content: 'Simple alert!' });

**Comment by Thirunav** [comment]
Hi,

confirm dialog is working fine now when I added the below jquery cdn into the page, but in another place i have used $(function(){ }) it's throwing uncaught error $ is not a function

<script
  src="https://code.jquery.com/jquery-3.2.1.min.js"
  integrity="sha256-hwg4gsxgFZhOsEEamdOYGBf13FyQuiTwlAQgxVSNgt4="
  crossorigin="anonymous"></script>

when I remove that line then it's throwing Uncaught TypeError: a(...).on is not a function ERROR

**Comment by craftpip** [comment]
jconfirm plugin must be loaded after jquery library, and your code should come after all that.
it was a dependency issue.

**Comment by Thirunav** [comment]
In jquery-confirm.min.js file the below code causes the Error:
Could you please help me to resolve this.

$(window).on("keydown",function(e){if(!keyDown){var $target=$(e.target);var pass=false;if($target.closest(".jconfirm-box").length){pass=true;}if(pass){$(window).trigger("jcKeyDown");}keyDown=true;}});$(window).on("keyup",function(e){keyDown=false;});})(jQuery,window);

**Comment by maxmas** [comment]
hello guys,
I am experiencing the same problem.
On my local server ( AppServ), it is no problem.
But when I upload to remote server, I got error: Uncaught TypeError: $.alert is not a function.

After a long period of time, I found the problem.
I used zepto.min.js in my page.
jconfirm plugin must be loaded before zepto.min.js library, 
Otherwise, it will generate error.
But I do not know, why in my local server, it will not.
